sqlite/vdbe.c

```diff
@@ -2495,7 +2495,7 @@ case OP_Ge: {             /* same as TK_GE, jump, in1, in3 */
     pOut = &aMem[pOp->p2];
     iCompare = res;
     res2 = res2!=0;  /* For this path res2 must be exactly 0 or 1 */
-    if( (pOp->p5 & SQLITE_KEEPNULL)!=0 ){
+    if( (pOp->p5 & SQLITE_KEEPNULL)==SQLITE_KEEPNULL ){
```

Comment:
No, original wont work if we are combining STOREP2 and JUMPIFNULL. If we reach here, STOREP2 is set already. If we use `!=0`, it would be always true.